core/src/main/java/com/amplitude/core/platform/intercept/IdentifyInterceptFileStorageHandler.kt

```diff
@@ -0,0 +1,118 @@
+package com.amplitude.core.platform.intercept
+
+import com.amplitude.common.Logger
+import com.amplitude.core.events.BaseEvent
+import com.amplitude.core.events.IdentifyOperation
+import com.amplitude.core.utilities.FileStorage
+import com.amplitude.core.utilities.toEvents
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.json.JSONArray
+import org.json.JSONException
+
+class IdentifyInterceptFileStorageHandler(
+    private val storage: FileStorage,
+    private val logger: Logger,
+    private val scope: CoroutineScope,
+    private val dispatcher: CoroutineDispatcher
+) : IdentifyInterceptStorageHandler {
+    override suspend fun getTransferIdentifyEvent(): BaseEvent? {
+        val eventsData = storage.readEventsContent()
+        if (eventsData.isEmpty()) {
+            return null
+        }
+        var event: BaseEvent? = null
+        var identifyEventUserProperties: MutableMap<String, Any?>? = null
+        for (eventPath in eventsData) {
+            try {
+                val eventsString = storage.getEventsString(eventPath)
+                if (eventsString.isEmpty()) continue
+                val eventsList = JSONArray(eventsString).toEvents()
+                var events = eventsList
+                if (event == null) {
+                    event = eventsList[0]
+                    identifyEventUserProperties = event.userProperties!![IdentifyOperation.SET.operationType] as MutableMap<String, Any?>
+                    events = eventsList.subList(0, eventsList.size)
+                }
+                val userProperties = IdentifyInterceptorUtil.mergeIdentifyList(events)
+                identifyEventUserProperties?.putAll(userProperties)
+                removeFile(eventPath as String)
+            } catch (e: JSONException) {
+                logger.warn("Identify Merge error: " + e.message)
+                removeFile(eventPath as String)
+            }
+            event?.userProperties!![IdentifyOperation.SET.operationType] =
+                identifyEventUserProperties
+        }
+        return event
+    }
+
+    override suspend fun fetchAndMergeToNormalEvent(event: BaseEvent): BaseEvent {
+        val eventsData = storage.readEventsContent()
+        if (eventsData.isEmpty()) {
+            return event
+        }
+        val userProperties = mutableMapOf<String, Any?>()
+        for (eventPath in eventsData) {
+            try {
+                val eventsString = storage.getEventsString(eventPath)
+                if (eventsString.isEmpty()) continue
```

Comment:
nit: single line if's are risky imo. Would prefer braces to prevent unexpected errors based on white space changes.
```suggestion
                if (eventsString.isEmpty()) {
                  continue
                }
```